### DISCUSSION TITLE
Keine Verbindung zur HB-RF-ETH mit Docker-Image

### DISCUSSION THREAD

**Discussion by thorsten-l** [OpenCCU/OpenCCU#1218] [🇩🇪 Fragen&Antworten]
**Describe the bug**
Keine Verbindung zur HB-RF-ETH mit Docker-Image

**Steps to reproduce the behavior**

Folgende Konfigs habe ich ausprobiert

piVCCU3 (Pi4+Raspbian), funktioniert mit HB-RF-ETH (erster Test)
Ubuntu 18.04 (amd64) + Raspberrymatic (Docker) - Keine Verbindung zur HB-RF-ETH
Raspbian (armhf) + Raspberrymatic (Docker) - Keine Verbindung zur HB-RF-ETH

Alle piVCCU Module sind geladen und der avahi daemon hat auch auf allen drei Sytemen die IP Adresse des HB-RF-ETH
angezeigt und in /etc/defaults/hb_rf_eth eingetragen. /dev/raw-uart und /dev/eq3loop exisitieren auf allen drei Systemen.
Damit sich die Systeme nicht gegenseitig stören, war immer nur eines der drei eingeschaltet.

** LOG **
```
Creating ccu ... done
Attaching to ccu
ccu               | Starting watchdog...
ccu               | Identifying onboard hardware: oci, OK
ccu               | Initializing RTC Clock: no hardware found
ccu               | Running sysctl: OK
ccu               | Checking for Factory Reset: not required
ccu               | Checking for Backup Restore: not required
ccu               | Initializing System: ln: /dev/raw-uart: File exists
ccu               | ln: /dev/eq3loop: File exists
ccu               | OK
ccu               | Starting logging: OK
ccu               | Populating /dev using udev: done
ccu               | Init onboard LEDs: init, OK
ccu               | Starting irqbalance: OK
ccu               | Starting network: eth0: link up, fixed, firewall, inet up, 172.19.0.2, OK
ccu               | Identifying Homematic RF-Hardware: ...HmRF: none, HmIP: none, OK
ccu               | Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
ccu               | Starting hs485dLoader: disabled
ccu               | Starting xinetd: OK
```

**System information:**
 - RaspberryMatic Version: 3.57.4.20210320
 - Used Hardware: RaspberryPi 4, Raspbian, 10 (buster) / Intel i7, Ubuntu 18.04.5 LTS
 - Used HomeMatic RF-Module: HB-RF-ETH + RPI-RF-MOD
 - Used Docker Version: 20.10.5, build 55c4c88

**Comment by thorsten-l** [comment]
Zwei Sachen sind mir aufgefallen sowohl `/dev/raw-uart` als auch `/dev/eq3loop` exisitieren bereits im Image, so das `/etc/init.d/S06InitSystem` diese nicht nach `/dev_host/...` als Symbolic-Link anlegen kann. Auch läuft auf der piVCCU3 ein monitor für die HB-RF-ETH Verbindung. `/var/lib/piVCCU3/monitor_hb_rf_connection.sh` diesen gibt es im Docker-Image nicht.

Ich habe `/etc/init.d/S06InitSystem` so verändert, das die Links nun angelegt werden, es gibt trozdem keine Verbindung 

```
  # setup /dev_host for OCI/Docker environments (excluding HA)
  if [[ "${HM_HOST}" == "oci" ]] && [[ -z "${HM_RUNNING_IN_HA}" ]]; then
    # To avoid sharing the whole host /dev with the container we
    # mount devtmps in /dev_host and symlink the required devices
    mount -o noatime -t devtmpfs devtmpfs /dev_host
    if [[ -d "/dev_host" ]]; then
      for device in raw-uart raw-uart1 eq3loop mmd_bidcos mmd_hmip usb; do
        if [[ -e "/dev_host/${device}" ]]; then
          if [[ -e "/dev/${device}" ]]; then
            mv "/dev/${device}" "/dev/${device}.local"
          fi
          ln -s "/dev_host/${device}" "/dev/${device}"
        fi
      done
    fi
  fi
```

**Comment by jens-maus** [comment]
Also als erstes ist das vermengen von piVCCU mit RaspberryMatic nie eine gute Idee. Wenn man also von piVCCU zu RaspberryMatic (auch als Docker) umsteigen will ist man gut beraten hier aufzuräumen und piVCCU gänzlich zu deaktivieren/installieren. Danach kann man dann wie hier angegeben vorgehen: https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI

Auch sehe ich nicht aus den Angaben ob auch die Datei `/etc/config/hb_rf_eth` wie in der Dokumentation zum HB-RF-ETH Betrieb mit RaspberryMatic (siehe https://github.com/jens-maus/RaspberryMatic/wiki/Experten-Features#hb-rf-eth-anbindung) gesetzt wurde. Die `/etc/defaults/hb_rf_eth` wie hier genannt ist nur bei der Nutzung mit piVCCU relevant.

**Comment by jens-maus** [comment]
Und da all dies kein Bug ist verwandle ich das ganze mal in ein Diskussionsbeitrag weil ich nicht sehe wie das auf einen Bug in RaspberryMatic hindeuten sollte.
